test/hpr_protocol_router_SUITE.erl

```diff
@@ -231,6 +233,77 @@ server_crash_test(_Config) ->
     ?assertEqual(0, erlang:length(hpr_protocol_router:all_streams())),
     ok.
 
+gateway_disconnect_test(_Config) ->
+    %% Startup Router server
+    {ok, ServerPid} = grpcbox:start_server(#{
+        grpc_opts => #{
+            service_protos => [packet_router_pb],
+            services => #{'helium.packet_router.packet' => hpr_test_packet_router_service}
+        },
+        listen_opts => #{port => 8082, ip => {0, 0, 0, 0}}
+    }),
+
+    %% Interceptor
+    Self = self(),
+    application:set_env(
+        hpr,
+        packet_service_route_fun,
+        fun(Env, StreamState) ->
+            {packet, Packet} = hpr_envelope_up:data(Env),
+            Self ! {packet_up, Packet},
+            StreamState
+        end
+    ),
+
+    {Route, EUIPairs, DevAddrRanges} = test_route(),
+    {ok, GatewayPid} = hpr_test_gateway:start(#{
+        forward => self(), route => Route, eui_pairs => EUIPairs, devaddr_ranges => DevAddrRanges
+    }),
+
+    %% Send packet and route directly through interface
+    ok = hpr_test_gateway:send_packet(GatewayPid, #{}),
+
+    PacketUp =
+        case hpr_test_gateway:receive_send_packet(GatewayPid) of
+            {ok, EnvUp} ->
+                {packet, PUp} = hpr_envelope_up:data(EnvUp),
+                PUp;
+            {error, timeout} ->
+                ct:fail(receive_send_packet)
+        end,
+
+    ok =
+        receive
+            {packet_up, RvcPacketUp} -> ?assertEqual(RvcPacketUp, PacketUp)
+        after timer:seconds(2) -> ct:fail(no_msg_rcvd)
+        end,
+
+    Gateway = hpr_test_gateway:pubkey_bin(GatewayPid),
+    LNS = hpr_route:lns(Route),
+    [{_, #{channel := ChannelPid, stream_pid := StreamPid}}] = ets:lookup(
+        hpr_protocol_router_ets, {Gateway, LNS}
+    ),
+
+    ok = gen_server:stop(GatewayPid),
+    timer:sleep(timer:seconds(2)),
```

Comment:
Does the timeout in the receive block not cover this?

Reply:
When a new stream is created it gets monitored every time look at line 116